### DISCUSSION TITLE
Cannot read properties of undefined (reading 'DT_FLOAT32')

### DISCUSSION THREAD

**Discussion by Pryme8** [donmccurdy/glTF-Transform#1516] [Q&A] [answered]
For some of my models when I run a meshopt encoder over the model I get this error:
```
model optimization failed TypeError: Cannot read properties of undefined (reading 'DT_FLOAT32')
[dev:api]     at initDecoderModule (D:\gitHUB\frame\frame\node_modules\@gltf-transform\extensions\src\khr-draco-mesh-compression\decoder.ts:94:49)
[dev:api]     at install (D:\gitHUB\frame\frame\node_modules\@gltf-transform\extensions\src\khr-draco-mesh-compression\draco-mesh-compression.ts:135:4)
[dev:api]     at Function.read (D:\gitHUB\frame\frame\node_modules\@gltf-transform\core\dist\index.modern.js:4345:21)
[dev:api]     at NodeIO.readJSON (D:\gitHUB\frame\frame\node_modules\@gltf-transform\core\dist\index.modern.js:5756:23)
[dev:api]     at NodeIO.read (D:\gitHUB\frame\frame\node_modules\@gltf-transform\core\dist\index.modern.js:5737:23)
[dev:api]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[dev:api]     at async <anonymous> (D:\gitHUB\frame\frame\api\modelOptimization\gltfTransform.ts:74:22)
```

Im not really sure how to troubleshoot this or what the correct steps are to check if this is going to happen on a model or not.

**Comment by donmccurdy** [comment, marked as answer]
It looks like the error occurs just when reading the model, not yet to the step of encoding with Meshopt, is that correct?

This is most likely a sign that the model uses Draco compression, and Draco dependencies are not installed, or not installed correctly. Documentation: 

https://gltf-transform.dev/modules/extensions/classes/KHRDracoMeshCompression

If the model doesn't use Draco compression, then no error should occur. That said ... there is supposed to be a clearer error message than this when the Draco encoder/decoder isn't installed, if you find that's not the case please let me know!

**Reply by Pryme8** [reply]
Thank you for your fast reply.  Ill see if that is the case, I thought it was already installed but its highly likely its not.

**Reply by Pryme8** [reply]
Thank you so much!  Looks like I deleted that import when moving to meshopt and did not realize it was a dependency.

**Reply by donmccurdy** [reply]
Awesome! You should be able to skip the _encoder_ if you're compressing with meshopt, but the decoder would still be needed when reading files already compressed with Draco.

If you're not yet, it might also be necessary also to delete any existing Draco extension before exporting with Meshopt compression. Using both is technically allowed... but one will take precedence, the other will be skipped, and this can lead to unpredictable results. 

Example:

https://github.com/donmccurdy/glTF-Transform/blob/aad01c370884ee1241cc969fb35948b39f4caa52/packages/cli/src/session.ts#L39-L49
